third_party/proton/proton/profile.py

```diff
@@ -215,6 +215,7 @@ def wrapper(*args, **kwargs):
         session = start(name, context=context, data=data, backend=backend, mode=mode, hook=hook)
         ret = func(*args, **kwargs)
         deactivate(session)
+        finalize(session)
```

Comment:
Avoid leaking the session. Otherwise, the session id is nowhere to be found. 

If it is expected to manually call `proton.finalize()` (with no arguments), I will revert this change.